mongo/index_view.go

```diff
@@ -52,7 +52,8 @@ type IndexModel struct {
 }
 
 func isNamespaceNotFoundError(err error) bool {
-	if de, ok := err.(driver.Error); ok {
+	var de driver.Error
+	if errors.As(err, &de) {
 		return de.Code == 26
```

Comment:
Optional: Can we use the `NamespaceNotFound` method here?
Optional: If we use `NamespaceNotFound`, can we remove this function?